onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -1344,8 +1346,16 @@ void ConvOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const Nod
   const auto& weight = input_defs[w_idx]->Name();
 
   const auto& weight_tensor = initializers.at(weight);
-  bool conv2d = (group == 1);
-  bool depthwise_conv2d = (weight_tensor.dims()[1] == 1);
+  bool conv_2d = false,
+       depthwise_conv_2d = false,
+       grouped_conv_2d = false;
+
+  if (group == 1)
+    conv_2d = true;
+  else if ((weight_tensor.dims()[1] == 1))
+    depthwise_conv_2d = true;
+  else
+    grouped_conv_2d = true;
 
   Shape onnx_weight_shape;
```

Comment:
Comment explaining the logic would be helpful.

Previously both conv2d and depthwise_conv2d could be set to true. Now only one can. Was the old code incorrect?

Reply:
The old code is correct since I always use !conv2d && depthwise_conv2d, also we ruled out the grouped_conv_2d in ConvOpBuilder::IsOpSupportedImpl, so when !conv2d -> depthwise_conv2d == ture


---
In reply to: [459171731](https://github.com/microsoft/onnxruntime/pull/4582#discussion_r459171731) [](ancestors = 459171731)